daft/dataframe/dataframe.py

```diff
@@ -285,7 +285,36 @@ def _from_tables(cls, *parts: MicroPartition) -> "DataFrame":
         builder = LogicalPlanBuilder.from_in_memory_scan(
             cache_entry, parts[0].schema(), result_pset.num_partitions(), size_bytes
         )
-        return cls(builder)
+
+        df = cls(builder)
+        df._result_cache = cache_entry
```

Comment:
Not sure if there would be a problem populating _result_cache, but I was thinking that this makes sense so users can also do `len(df)`